20231018-access-logs.md

```diff
@@ -0,0 +1,188 @@
+
+#### Access logs in Storj Private Cloud
+
+![SPC access logs](20231018-access-logs/access-logs-spc.drawio.svg "Storj Private Cloud Access Logs")
+
+Storj Private Cloud can come in many different setups. We can implement delayed and real-time logs with the open-source Loki server for customers who prefer to have the complete installation open-source.
+
+We also have the option to use [Grafana Enterprise Logs (GEL)](https://grafana.com/docs/enterprise-logs/latest/) for more complex setups where the customer would be willing to pay for the GEL license and support.
+
+#### Configuring a bucket for access logs
+
+The customer will be able to turn on access logs per bucket. By default, a bucket does not generate access logs. The customer may decide later to turn off the access logs for the bucket.
+
+Initially, turning access logs on and off will be requested with a support ticket. Designing and implementing a UI will be the subject of a future design doc.
+
+The customer will select a source bucket to generate access logs and a target bucket to receive the delayed access logs. The customer will create S3 credentials with full permission to the target bucket. The Storj Loki server will configure these S3 credentials as a target object store for the Tenant ID representing the source bucket.
+
+The target bucket can be either in the same Storj project as the source bucket, a different Storj project, or even a different Storj account. Choosing the source bucket as the target bucket is not recommended, as it will cause an infinite loop of access log generation.
+
+Customers who want to configure log retention for the delayed access logs can configure the [S3 credentials with Object TTL](https://docs.storj.io/dcs/buckets/object-lifecycles#setting-object-ttl-in-the-access-grant-s3-credentials) for the desired retention period.
+
+Customers who want to turn on real-time access logs will provide the Push URL to their Grafana Cloud stack to receive the logs. The Push URL will include their Grafana Cloud user ID and API token. The logging agent will use the Push URL as a destination for the generated access log for the configured Storj bucket.
+
+## Disclaimers
+
+### Anti-goals
+
+- Easy to configure.
+  - Initially, it's okay for customers to open a support ticket to enable access logs.
+  - No need to think about UI yet.
+- Log targets outside of Storj.
+  - We are a storage company. Hence, we want to store the access logs in Storj buckets. This way, storing and querying the logs implicitly incur usage of our product so that we can provide this feature without additional fees.
+  - We may allow non-Storj targets for specific use cases like Grafana Cloud for real-time logs. It could be a premium feature for an additional fee.
+
+### Alternatives considered
+
+- Implement it all in-house.
+
+### Open question
+
+- Should we log linksharing requests beyond those to raw content like listing buckets and prefixes, displaying the object map, etc.?
```

Comment:
I think that would be useful, maybe we could make it clear where these are coming from with the user agent.